### PR TITLE
Add reporting functions tests and update coverage plan

### PR DESCRIPTION
**Pull request by zuhairmahd**
- Add comprehensive unit tests for reporting functions:
  - tests/Unit/reportingFunctions/AssessDeviceState.Tests.ps1 (full NextUserReadiness coverage, 25+ scenarios)
  - tests/Unit/reportingFunctions/Export-DeviceList.Tests.ps1 (autopilot/imported/unmanaged/managed, file modes, error handling)
  - Expand tests/Unit/ShowDeviceReport.Tests.ps1 to cover report building, property formatting, menu integration, null/edge cases and logging

- Update docs/COVERAGE_IMPROVEMENT_PLAN.md:
  - Bump Document Version to 2.0 and Last Updated to Nov 4, 2025
  - Reflect Week 9 completion (reporting functions), +81 tests, total 1,278 passing tests, estimated ~45% coverage
  - Add Week 9 deliverables, execution metrics, technical notes and identified production issue (connectToNetwork missing from strings.psd1)

Notes:
- Added stubs/mocks (Export-Csv, Export-DeviceReport) and targeted mocks for PS 5.1/7 compatibility and to avoid infinite menu loops during tests.
- Focus on improving coverage and stabilizing reporting-related code paths.

### CODE REVIEW

File: tests/Unit/reportingFunctions/AssessDeviceState.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+            ready    = 'The device is ready for the next user'
+            notReady = 'The device is not ready for the next user'
+        }
+        
+        $script:deviceActions = @{
+            contactAdmin     = 'Contact an Intune administrator'
+            contactHelpdesk  = 'Contact the helpdesk'
+            WipeOrClean      = 'Wipe or clean the device'
+            none             = 'No action'
+            connectToNetwork = 'Connect to network'  # Note: Not in strings.psd1 but referenced in AssessDeviceState.ps1 line 177
```
[the remark this answers is not in the document]
Comment (zuhairmahd) [reply tests/Unit/reportingFunctions/AssessDeviceState.Tests.ps1]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/zuhairmahd/Autopilot/pull/183#discussion_r2492600179)
